external/loaders/loaders/mappers/_nudged/_nudged.py

```diff
@@ -230,3 +227,70 @@ def _get_datasets(
         ds = xr.open_zarr(mapper, consolidated=consolidated)
         datasets[source] = ds
     return datasets
+
+
+@mapper_functions.register
+def open_nudge_to_fine_scream(
+    data_path: str,
+    nudging_variables: Sequence[str],
+    physics_timestep_seconds: float = 900.0,
+    consolidated: bool = True,
+    datasets: Sequence[str] = (
+        "physics_tendencies.zarr",
+        "nudging_tendencies.zarr",
+        "state_after_timestep.zarr",
+    ),
+    cache_size_mb: Optional[float] = None,
+) -> XarrayMapper:
+    """
+    Similar to open_nudge_to_fine, but for scream specifically.
+
+    Args:
+        url (str):  path to nudge-to-fine output directory, remote or local
+        nudging_variables (Sequence[str]): Names of nudged variables, nudging tendency
+            will be subtracted to retrieve model state before nudging
+        physics_timestep_seconds (float): physics timestep, i.e., dt_atmos; defaults
+            to 900.0
+        consolidated (bool): whether zarrs to open have consolidated metadata
+        datasets: names of zarrs at the given URL to include, defaults are
+            physics_tendencies.zarr, nudging_tendencies.zarr, and
+            state_after_timestep.zarr (which you should probably include).
+            For example, you may want to include also "diags.zarr" to retrieve
+            total_precipitation_rate.
+         cache_size_mb: Cache size in MB for using zarr.storage.LRUStoreCache
+            for accessing data. A cache of this size is created for each zarr
+            dataset in the datasets arg. No LRU caches created if this arg is not
+            supplied.
+
+    Returns:
+        mapper to dataset containing nudging tendencies, physics tendencies,
+            and model state data
+    """
+
+    ds = xr.merge(
+        _get_datasets(
+            data_path, datasets, consolidated=consolidated, cache_size_mb=cache_size_mb
+        ).values(),
+        join="inner",
+    )
+
+    differenced_state: Dataset = {}
+    for nudging_variable in nudging_variables:
+        nudging_tendency = ds[f"{nudging_variable}_tendency_due_to_nudging"]
+        differenced_state[nudging_variable] = (
+            ds[nudging_variable] - nudging_tendency * physics_timestep_seconds
+        )
+    ds = ds.assign(differenced_state)
```

Comment:
Do you know where in the time loop nudging happens for SCREAM? The purpose of this code is to make sure that the ML inputs are the atmospheric state right before nudging. It's possible this code will need to change if the nudging is implemented differently than what we have in FV3GFS.

Reply:
Right, it is actually flexible in scream where nudging happens (the order of any process is specified in the namelist). I'm keeping it like this because we do not actually have a nudged run yet, but I added a todo note to update this once we have it solidified.